### PR TITLE
Improve unicode handling in JS sources

### PR DESCRIPTION
**Pull request by dcramer**
- correct various unicode support for sources/maps
- add test for embedded sourcemaps
- add test for fetch_release_file

/cc @getsentry/infrastructure

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/3878)

### CODE REVIEW

File: src/sentry/lang/javascript/errormapping.py [hunk 1]
```diff
@@ -67,6 +67,8 @@ def load_mapping(self):
         return data
 
     def try_process(self, exc):
+        if not exc['value']:
```
Comment (dcramer) [inline src/sentry/lang/javascript/errormapping.py]: unrelated, i just saw this showing up in a bunch of tests and was tired of it

File: src/sentry/lang/javascript/processor.py [hunk 2]
```diff
@@ -253,7 +254,7 @@ def fetch_release_file(filename, release):
         else:
             # Write the compressed version to cache, but return the deflated version
             cache.set(cache_key, (releasefile.file.headers, z_body, 200), 3600)
-            result = (releasefile.file.headers, body, 200)
+            result = (releasefile.file.headers, body.decode('utf-8'), 200)
```
Comment (dcramer) [inline src/sentry/lang/javascript/processor.py]: the changes to fetch_release_file are the primary current bug

requests returns unicode, fetch_release_file was returning bytes

File: tests/sentry/lang/javascript/test_processor.py [hunk 3]
```diff
@@ -12,17 +12,59 @@
...
 from sentry.lang.javascript.errormapping import (
     rewrite_exception, REACT_MAPPING_URL
 )
-from sentry.models import Release
+from sentry.models import File, Release, ReleaseFile
 from sentry.testutils import TestCase
 
 base64_sourcemap = 'data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiZ2VuZXJhdGVkLmpzIiwic291cmNlcyI6WyIvdGVzdC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiO0FBQUEiLCJzb3VyY2VzQ29udGVudCI6WyJjb25zb2xlLmxvZyhcImhlbGxvLCBXb3JsZCFcIikiXX0='
 
+unicode_body = u"""function add(a, b) {
```
Comment (dcramer) [inline tests/sentry/lang/javascript/test_processor.py]: i would have used TestCase.load_fixture, but currently JS special cases the path to its fixtures and I didn't want to move them all

File: tests/sentry/lang/javascript/test_plugin.py [hunk 4]
```diff
@@ -271,8 +277,10 @@ def test_sourcemap_source_expansion(self):
             'function add(a, b) {',
             '\t"use strict";',
         ]
-        assert frame.context_line == '\treturn a + b;'
-        assert frame.post_context == ['}']
+        expected = u'\treturn a + b; // fôo'
+        print(repr(frame.context_line))
```
Comment (mattrobenolt) [inline tests/sentry/lang/javascript/test_plugin.py]: print

**Pull request merged**